server/events/project_finder.go

```diff
@@ -167,7 +167,11 @@ func (p *DefaultProjectFinder) DetermineProjects(log logging.SimpleLogging, modi
 	exists := p.removeNonExistingDirs(uniqueDirs, absRepoDir)
 
 	for _, p := range exists {
-		projects = append(projects, models.NewProject(repoFullName, p))
+		// It's unclear how we are supposed to determine the project name at this point
+		// For now, we'll just add the default projectName
+		// TODO: Add support for non-default projectName
+		projectName := ""
+		projects = append(projects, models.NewProject(repoFullName, p, projectName))
```

Comment:
in a case like this : ```
- name: pepe-ue2-atlantis-acm-cert
  workspace: pepe-ue2-atlantis-acm-cert
  workflow: atlantisworkflow
  dir: components/terraform/acm
  terraform_version: v1.4.6
  delete_source_branch_on_merge: true
  autoplan:
    enabled: false
    when_modified:
    - '**/*.tf'
    - $PROJECT_NAME.tfvars.json```

The workspace will be used for the checkout dir path and I wonder if work too?

Reply:
Adding a ProjectName of `""` to a Project should be identical to a Project that previously had no ProjectName set at all. I agree that eventually getting the "correct" ProjectName into Project is a good thing, but it's no worse than before. Additionally, the only thing ProjectName is currently used for is deleting the plan file, so I can't see how this would make anything worse.

I do believe we should follow up this PR with a more holistic view of the role of ProjectNames (I'm gathering a list of TODOs)